### PR TITLE
v2.13.0: feat(): Rearranged endpoints to match new API Portal

### PR DESCRIPTION
**Pull request by JJ-Cro**
New api portal:
https://developers.binance.com/en

No functional changes in this PR, just moving methods within each class and some deprecation markers as jsdoc

### CODE REVIEW

File: src/types/spot.ts [hunk 1]
```diff
@@ -1552,8 +1557,7 @@ export interface SubAccountEnableLeverageToken {
   enableBlvt: boolean;
 }
 
-export interface EnableOrDisableIPRestrictionForSubAccountParams
-  extends BasicSubAccount {
+export interface C extends BasicSubAccount {
```
Comment (tiagosiebler) [inline src/types/spot.ts:1560]: Is this a typo? `interface C`? looks like an accidental deletion

File: src/main-client.ts [hunk 2]
```diff
@@ -1177,2206 +657,63 @@ export class MainClient extends BaseRestClient {
...
-      urlSuffix += '?symbols=' + symbols;
-    }
-
-    return this.get('api/v3/exchangeInfo' + urlSuffix);
-  }
-
-  getOrderBook(params: OrderBookParams): Promise<OrderBookResponse> {
-    return this.get('api/v3/depth', params);
+  // spot
+  getSpotUserDataListenKey(): Promise<{ listenKey: string }> {
```
Comment (tiagosiebler) [inline src/main-client.ts:665]: You can include all of these listen key endpoints as one "category" in the v2 main client. Same for futures and the futures v2 client. Easier to see if they're all grouped together

File: src/main-client.ts [hunk 3]
```diff
@@ -522,526 +522,6 @@ export class MainClient extends BaseRestClient {
...
-    managerSubTransferHistoryVos: ManagerSubTransferHistoryVos[];
-    count: number;
-  }> {
-    return this.getPrivate(
-      'sapi/v1/managed-subaccount/query-trans-log',
-      params,
-    );
-  }
-
   /**
```
Comment (tiagosiebler) [inline src/main-client.ts:525]: For the broker (link) endpoints, these are in the docs here:
https://developers.binance.com/docs/binance_link/Introduction

Some might be unofficial but usually have a doc. Let me know if there's any you're not sure about that aren't in the docs.

File: src/main-client.ts (hunk 2 above)
Comment (JJ-Cro) [reply src/main-client.ts]: Included.

File: src/main-client-v2.ts [hunk 4]
```diff
@@ -3190,6 +3191,328 @@ export class MainClient extends BaseRestClient {
     return this.getPrivate('sapi/v1/pay/transactions', params);
   }
 
+  /**
```
Comment (JJ-Cro) [inline src/main-client-v2.ts:3194]: This is where new part  of broker API starts.  It is now called Exchange link, as you saw. 

Note: I added some of the missing endpoints, but only some. THere is 10 or so missing, I will take care of that later when we are done. with this rearranging.

File: src/main-client-v2.ts [hunk 5]
```diff
@@ -3190,6 +3191,328 @@ export class MainClient extends BaseRestClient {
...
+    if (days === 30) {
+      return this.getPrivate('sapi/v1/apiReferral/rebate/historicalRecord', {
+        customerId,
+      });
+    }
+  }
+
+  /**
+   * Broker Endpoints - only on old docs
+   */
```
Comment (JJ-Cro) [inline src/main-client-v2.ts:3386]: Here are marked broker endpoints that are no longer in new docs. All broker urls now are under sapi/vx/broker/smth 
, I dont see this apiReferral, only in old docs. So I marked them, until all old docs become deprecated, I guess we can keep it like that.

File: src/main-client-v2.ts [hunk 6]
```diff
@@ -3190,6 +3191,328 @@ export class MainClient extends BaseRestClient {
...
+    if (!params[orderIdProperty].startsWith(expectedOrderIdPrefix)) {
+      logInvalidOrderId(orderIdProperty, expectedOrderIdPrefix, params);
+    }
+  }
+
+  /**
+   *
+   * User Data Stream Endpoints
+   *
+   **/
```
Comment (JJ-Cro) [inline src/main-client-v2.ts:3461]: Here I added User Data stream, grouped as they were. Will also double check if we are missing any by going through all of them  after this is done :)

File: src/main-client-v2.ts [hunk 7]
```diff
@@ -3362,30 +3685,6 @@ export class MainClient extends BaseRestClient {
       params,
     );
   }
-
-  /**
```
Comment (JJ-Cro) [inline src/main-client-v2.ts]: This is not missing, I moved it up few lines, above the data stream. Makes more sense to be there I guess

File: src/main-client.ts [hunk 8]
```diff
@@ -521,199 +521,4 @@ export class MainClient extends BaseRestClient {
     const response = await this.getForBaseUrl(endpoint, baseUrlKey);
     return response.serverTime;
   }
-
-  /**
-   * Broker Endpoints
-   */
-
-  getBrokerIfNewSpotUser(): Promise<{
```
Comment (JJ-Cro) [inline src/main-client.ts:1049]: Moved all of these to new v2. More deets in v2 file

File: src/usdm-client.ts [hunk 9]
```diff
@@ -113,131 +113,4 @@ export class USDMClient extends BaseRestClient {
...
-    return this.getPrivate('fapi/v1/pmAccountInfo', params);
-  }
-
-  /**
-   *
-   * Broker Futures Endpoints
-   *
-   **/
-
-  // 1 == USDT-Margined, 2 == Coin-margined
```
Comment (JJ-Cro) [inline src/usdm-client.ts:567]: Moved these to v2, but I think they might soon be obsolete - as I saw, all broker endpoints are using /broker/ url, so I guess this wont work anymore. I didnt test, we can check later and mark deprecated. ANyways, all of it is in v2, at the bottom :)

File: src/usdm-client.ts [hunk 10]
```diff
@@ -113,131 +113,4 @@ export class USDMClient extends BaseRestClient {
...
-
-  /**
-   *
-   * User Data Stream Endpoints
-   *
-   **/
-
-  // USD-M Futures
-
-  getFuturesUserDataListenKey(): Promise<{ listenKey: string }> {
```
Comment (JJ-Cro) [inline src/usdm-client.ts:664]: Moved it all as a group.

File: src/coinm-client.ts [hunk 11]
```diff
@@ -116,116 +116,7 @@ export class CoinMClient extends BaseRestClient {
...
   }
 
-  /**
-   *
-   * Broker Futures Endpoints
-   *
-   **/
-
-  // 1 == USDT-Margined, 2 == Coin-margined
-  getBrokerIfNewFuturesUser(
```
Comment (JJ-Cro) [inline src/coinm-client.ts:465]: All broker stuff, as marked in all clients, now seems to be /broker/ - you can read more in other files.

**Review by tiagosiebler — COMMENTED** [review]
Main client looks good, just a few deprecation comments that could point to the newer functions

**Review by tiagosiebler — APPROVED** [review]
LGTM, thanks!

**Pull request merged**